langchain4j-mongodb-atlas/src/test/java/dev/langchain4j/store/embedding/mongodb/MongoDBEmbeddingStoreCloudIT.java

```diff
@@ -0,0 +1,78 @@
+package dev.langchain4j.store.embedding.mongodb;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.AllMiniLmL6V2QuantizedEmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.EmbeddingStoreIT;
+import lombok.SneakyThrows;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.PojoCodecProvider;
+import org.bson.conversions.Bson;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+@Disabled("Need Cloud Mongo Atlas Credential")
+public class MongoDBEmbeddingStoreCloudIT extends EmbeddingStoreIT {
+
+    static MongoClient client;
+
+    MongoDBEmbeddingStore embeddingStore = MongoDBEmbeddingStore.builder()
+            .fromClient(client)
+            .databaseName("test_database")
+            .collectionName("test_collection")
+            .indexName("test_index")
+            .build();
+
+    EmbeddingModel embeddingModel = new AllMiniLmL6V2QuantizedEmbeddingModel();
+
+    @BeforeAll
+    static void beforeAll() {
+        client = MongoClients.create("mongodb+srv://<username>:<password>@<host>/?retryWrites=true&w=majority");
```

Comment:
I have tried running this test with my username/password/host, but getting this error:

`com.mongodb.MongoCommandException: Command failed with error 8000 (AtlasError): '$listSearchIndexes is not allowed or the syntax is incorrect, see the Atlas documentation for more information' on server ac-gpxo1fa-shard-00-02.1x5hb8d.mongodb.net:27017. The full response is {"ok": 0, "errmsg": "$listSearchIndexes is not allowed or the syntax is incorrect, see the Atlas documentation for more information", "code": 8000, "codeName": "AtlasError"}`

Reply:
Sorry about that, will fix after handle my network problem

Reply:
No worries, maybe the problem is on my side

Reply:
hmmmm, after reading the document, it seems that `$listSearchIndexes` requires an Atlas cluster tier of at least M10(see [docs](https://www.mongodb.com/docs/manual/reference/operator/aggregation/listSearchIndexes/), which means this opearation is not free in the cloud). Maybe we should let user create collection and index themselves using UI or other ways? What do you think?

Reply:
Another way to solve it is to add a attribute named `prod`, which means whether to create index, if it is set to true, then user should create index themselves, otherwise `MongoDBEmbeddingStore` will create index automatically.

This is the way I'm using right now

Reply:
Done